cairo/src/font/user_fonts.rs

```diff
@@ -0,0 +1,205 @@
+//pub fn cairo_user_font_face_set_init_func(
+use super::{FontExtents, FontFace, ScaledFont, TextCluster, TextClusterFlags, TextExtents};
+use crate::utils::status_to_result;
+use crate::{Context, Error, Glyph};
+
+pub struct UserFontFace(FontFace);
+
+impl UserFontFace {
+    #[doc(alias = "cairo_user_font_face_create")]
+    pub fn create() -> Result<FontFace, Error> {
+        let font_face = unsafe { FontFace::from_raw_full(ffi::cairo_user_font_face_create()) };
+        let status = unsafe { ffi::cairo_font_face_status(font_face.to_raw_none()) };
+        status_to_result(status)?;
+        Ok(font_face)
+    }
+
+    #[doc(alias = "cairo_user_font_face_set_init_func")]
+    pub fn set_init_func<F>(&self, func: F)
+    where
+        F: Fn(&mut ScaledFont, &Context) -> Result<FontExtents, Error> + Send + Sync + 'static,
```

Comment:
Does this have to be a mutable reference? All the API on `ScaledFont` is using `&self`